utility/servicer/module.go

```diff
@@ -182,31 +356,107 @@ func (s servicer) admitRelay(relay *coreTypes.Relay) error {
-		return fmt.Errorf("%s: relay failed servicer instance validation: %w", errPrefix, err)
+	if err := s.validateApplication(session); err != nil {
+		return fmt.Errorf("%s: %s: %w", errPrefix, err.Error(), errValidateApplication)
 	}
 
 	return nil
 }
 
+// DISCUSS: do we need to export this functionality as part of the utility module?
+// calculateAppSessionTokens determines the number of "session tokens" an application gets at the beginning
+// of every session. Each servicer will serve a maximum of ~(Session Tokens / Number of Servicers in the Session) relays for the application
+func (s *servicer) calculateAppSessionTokens(session *coreTypes.Session) (*big.Int, error) {
+	appStake, err := utils.StringToBigInt(session.Application.StakedAmount)
+	if err != nil {
+		return nil, fmt.Errorf("Error processing application's staked amount %s: %w", session.Application.StakedAmount, coreTypes.ErrStringToBigInt(err))
+	}
+
+	// DOCUMENT: find the right document to explain the following:
+	//	We assume that the value of certain parameters only changes/takes effect at the start of a session.
+	//	In this specific case, the `AppSessionTokensMultiplierParamName` parameter is retrieved for the height that
+	//		matches the beginning of the session.
+	readCtx, err := s.GetBus().GetPersistenceModule().NewReadContext(session.SessionHeight)
+	if err != nil {
+		return nil, fmt.Errorf("error getting persistence context at height %d: %w", session.SessionHeight, err)
+	}
+	defer readCtx.Release() //nolint:errcheck // We only need to make sure the readCtx is released
+
+	// DISCUSS: using an interface for returning each defined parameter seems less error-prone: also could return e.g. int64 in this case to remove the type cast
+	appStakeTokensMultiplier, err := readCtx.GetIntParam(typesUtil.AppSessionTokensMultiplierParamName, session.SessionHeight)
+	if err != nil {
+		return nil, fmt.Errorf("error reading parameter %s at height %d from persistence: %w", typesUtil.AppSessionTokensMultiplierParamName, session.SessionHeight, err)
+	}
+
+	return appStake.Mul(appStake, big.NewInt(int64(appStakeTokensMultiplier))), nil
+}
+
+// executeHTTPRequest performs the HTTP request that sends the relay to the chain's URL.
+func (s *servicer) executeHTTPRelay(meta *coreTypes.RelayMeta, payload *coreTypes.JsonRpcPayload) (*coreTypes.RelayResponse, error) {
+	if meta == nil || meta.RelayChain == nil || meta.RelayChain.Id == "" {
+		return nil, fmt.Errorf("Relay for application %s does not specify relay chain", meta.ApplicationAddress)
+	}
+
+	serviceConfig, ok := s.config.Services[meta.RelayChain.Id]
+	if !ok {
+		return nil, fmt.Errorf("Chain %s not found in servicer configuration: %w", meta.RelayChain.Id, errValidateRelayMeta)
+	}
+
+	chainUrl, err := url.Parse(serviceConfig.Url)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing chain URL %s: %w", serviceConfig.Url, err)
+	}
+	targetUrl := chainUrl.JoinPath(payload.HttpPath)
```

Comment:
Can you add a comment on what this is doing? A small example in the comment above would help

Reply:
Thanks for the review. This is not needed anymore: we have removed `HttpPath` from `JSONRPC` relays' payloads. Also fixed an issue with the HTTP call below: it should use POST: the `Method` on the JSONRPC is not an HTTP method.